sphinx-docs/screenshot.py

```diff
@@ -283,21 +281,33 @@ def _command_handler(self, commands):
                          "notes": "",
                        }        
         from_str_arg['inputs'] = reduce(lambda x,y: x+y, map(_cmd_to_inputs, commands), [])
-        from_str_arg["inputs"].append({"<slug>":self.filename})
-        if hasattr(self, "focus_selector"):
-            from_str_arg["focus"] = {}
-            from_str_arg["focus"]["selector"] = self.focus_selector
-            from_str_arg["focus"]["styles"] = FOCUS_CSS_STYLES
-            from_str_arg["notes"] = self.focus_annotation if hasattr(self, "focus_annotation") else ""
+        from_str_arg = self._common_arg_prep(from_str_arg)
         self.env.screenshot_all_screenshots.append({
             'docname':  self.env.docname,
             'from_str_arg': from_str_arg,
         })
+        return self._make_image_node()
+
+    def _common_arg_prep(self, arg):
+        """All commands are handled in the same way regarding these options. """
+        new_arg = arg
+        new_arg["inputs"].append({"<slug>":self.filename})
+        if hasattr(self, "focus_selector"):
+            new_arg["focus"] = {}
+            new_arg["focus"]["selector"] = self.focus_selector
+            new_arg["focus"]["styles"] = FOCUS_CSS_STYLES
+            new_arg["notes"] = self.focus_annotation if hasattr(self, "focus_annotation") else ""
+        new_arg["registered"] = self.options["registered"]
+        return new_arg
+
+    def _make_image_node(self):
+        """Make an image node by safely calling Image.run (i.e. ensure the file exists)."""
         self.arguments.append(os.path.join("/", SS_DUMP_DIR, self.filename+".png"))
         open(os.path.join(OUTPUT_PATH, self.filename+".png"), 'w').close()
```

Comment:
Hmm strange, we open the file and close it immediately? Is this a `touch`? Think we have an idiom for that already from the previous review.